thinc/loss.py

```diff
@@ -143,6 +148,25 @@ def get_grad(self, guesses: Floats2d, truths: IntsOrFloatsOrStrs) -> Floats2d:
             raise ValueError(err)
         difference = guesses - target
         difference *= mask
+        # Weight samples proportinal to the marginals.
+        # For each sample the target labels are normalized
+        # to sum to 1 and the normalized-labels are used
+        # to compute the weighted average of class_weights
+        if self.class_weights is not None:
+            if target.shape[1] != len(self.class_weights):
+                raise ValueError(
+                    "The number of classes in the "
+                    "target and in class weights "
+                    "has to be equal, but found "
+                    f"{target.shape[1]} and "
+                    f"{len(self.class_weights)}"
+                )
+            cp = xp.asarray(self.class_weights)
+            cp = xp.tile(cp, (guesses.shape[0], 1))
```

Comment:
I don't think this is really necessary right? The `cp` array can just be broadcast in

```
sample_weights = (norm_target * cp).sum(axis=1)
```

I think it already works without any changes, but otherwise I think

```
cp = xp.asarray(self.class_weights).reshape(1, target.shape[1])
```

should do the trick? Avoids a potentially large allocation :).